packages/tools-iottwinmaker/src/lib/workspace.ts

```diff
@@ -0,0 +1,182 @@
+  console.log('Creating S3 Bucket for the TwinMaker Workspace...');
+  const createBucketResponse: CreateBucketCommandOutput = await aws().s3.createBucket({ Bucket: s3BucketName });
+  console.log(`Workspace S3 Bucket created: ${createBucketResponse.Location}`);
+  console.log('Configuring CORS policy...');
+  aws().s3.putBucketCors({
+    Bucket: s3BucketName,
+    CORSConfiguration: {
+      CORSRules: [
+        {
+          AllowedHeaders: ['*'],
+          AllowedMethods: ['GET', 'PUT', 'POST', 'DELETE', 'HEAD'],
+          AllowedOrigins: ['*'],
+          ExposeHeaders: ['ETag', 'Access-Control-Allow-Origin'],
+        },
+      ],
+    },
+  });
+  return { s3BucketName, s3BucketArn: `arn:aws:s3:::${s3BucketName}` };
+}
+
+/**
+ * Driver function for workspace creation
+ * @param workspaceId workspace-id to be created
+ * @returns promise of workspace arn, s3 arn, role arn, dashboardrole arn
+ */
+async function prepareWorkspace(workspaceId: string) {
+  const uniqueId: string = createUniqueId();
+  const region: string = aws().region;
+  const { accountId, accountArn } = await aws().getCurrentIdentity();
+  const { s3BucketArn } = await createWorkspaceS3Bucket(uniqueId);
+
+  const policyParams: Record<string, string> = {
+    accountId,
+    accountArn,
+    region,
+    workspaceS3BucketArn: s3BucketArn,
+  };
+
+  const workspaceRolePolicy: string = replaceTemplateVars(JSON.stringify(workspaceRolePolicyTemplate), policyParams);
+  const workspaceRoleArn: string = await createRoleAndPolicy(
+    uniqueId,
+    'WorkspaceRole',
+    JSON.stringify(WORKSPACE_ROLE_ASSUME_POLICY),
+    workspaceRolePolicy
+  );
+
+  const workspaceArn: string = await retryFunction(async () => {
+    const createWorkspaceResponse: CreateWorkspaceCommandOutput = await aws().tm.createWorkspace({
+      workspaceId,
+      role: workspaceRoleArn,
+      s3Location: s3BucketArn,
+    });
+    const workspaceArn: string | undefined = createWorkspaceResponse.arn;
+    if (!workspaceArn) {
+      throw new Error('Failed to create Workspace');
+    }
+    console.log(`Created Workspace: ${workspaceArn}`);
+    return workspaceArn;
+  }, 10);
+
+  policyParams.workspaceArn = workspaceArn;
+  policyParams.dashboardRoleAssumedByArn = accountArn;
+  const workspaceDashboardRoleAssumePolicy: string = replaceTemplateVars(
+    JSON.stringify(WORKSPACE_DASHBOARD_ROLE_ASSUME_POLICY),
+    policyParams
+  );
+  const workspaceDashboardRolePolicy: string = replaceTemplateVars(
+    JSON.stringify(workspaceDashboardRolePolicyTemplate),
+    policyParams
+  );
+  const workspaceDashboardRoleArn: string = await createRoleAndPolicy(
+    uniqueId,
+    'WorkspaceDashboardRole',
+    workspaceDashboardRoleAssumePolicy,
+    workspaceDashboardRolePolicy
+  );
+
+  return {
+    workspaceArn,
+    workspaceS3BucketArn: s3BucketArn,
+    workspaceRoleArn,
+    workspaceDashboardRoleArn,
+  };
+}
+
+/**
+ * Wrapper function for workspace creation
```

Comment:
Nit: I don't think this is a wrapper function, documentation could just be:
`Creates a workspace if it does not already exist`